sootup.java.bytecode/src/test/java/sootup/java/bytecode/interceptors/typeresolving/TypeResolverTest.java

```diff
@@ -294,6 +293,39 @@ public void testStringDefaultMethodsTest() {
         new Local("l1", PrimitiveType.getBoolean()));
   }
 
+  @Test
+  public void testTAWarningWithoutRTJar() {
+    String directoryPath = baseDir + "Misc/";
+    try {
+      List<String> jarFileNames = listJarFiles(directoryPath);
+      jarFileNames.forEach(
+          jarFile -> {
+            System.out.println("Applying Type Assigner to jar " + jarFile);
+            applyTypeAssignerToJar(jarFile);
+          });
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public static List<String> listJarFiles(String directoryPath) throws IOException {
+    try (Stream<Path> paths = Files.walk(Paths.get(directoryPath))) {
+      return paths
+          .filter(Files::isRegularFile)
```

Comment:
Can we, instead of using a jar for the test case, use a source/class file pair so that one can check the corresponding source?